src/plugins/intel_gpu/src/kernel_selector/cl_kernels/fully_connected_gpu_bf_tiled_dyn_quan.cl

```diff
@@ -0,0 +1,803 @@
+    // To do this solve first input feature separately.
+    {
+        INPUT0_TYPE tmp_input = input[input_offset + get_sub_group_local_id() % TILE_B * TILE_IN_B_PITCH];
+        ACCUMULATOR_VEC_TYPE tmp_wei = TO_ACCUMULATOR_VEC_TYPE(BLOCK_READN(FILTER_TYPE, TILE_OFM, weights, weights_offset));
+        #if COMPRESSED_WEIGHTS
+            tmp_wei = (tmp_wei - d_zp) * d_scale;
+        #endif
+        unroll_for(uint bi = 0; bi < TILE_B; ++bi) {
+            acc[bi] = _sub_group_shuffle(tmp_input, bi) * tmp_wei;
+        }
+
+        weights_offset += TILE_OFM * SIMD;
+        input_offset += 1;
+    }
+#endif
+
+    // =====================================================================================================================================
+    // Main computation loop
+    uint iterations = MAIN_LOOP_ELEMENTS_COUNT / (TILE_IFM * SIMD);  // TEMP : iterations = 4096 / (2 * 16);
+    __attribute__((opencl_unroll_hint(1)))
+    for (uint ni = 0; ni < iterations; ++ni) {
+        // Load input.
+        #define LOAD_IN_0(bi) do {                                    \
+                in_0[bi] = INPUT_BLOCK_READ(input, input_offset);     \
+                input_offset += TILE_IN_B_PITCH;                      \
+            } while (false)
+
+        CONST_LOOP(TILE_B, LOAD_IN_0);
+        #undef LOAD_IN_0
+
+        input_offset += TILE_IFM * SIMD - TILE_IN_B_PITCH * TILE_B;
+        // NOTE: Manually unrolling multiplication loop leads to lower register pressure and allows for bigger block sizes,
+        //       but significantly degrades readability and generality of code.
+        //       It doesn't also show noticable performance improvement on tested configurations.
+        #if DECOMPRESSION_SCALE_POST_OP
+            // ACCUMULATOR_VEC_TYPE acc_tmp[TILE_B] = { };
+            MAKE_VECTOR_TYPE(int, TILE_OFM) acc_tmp[TILE_B] = { };
+            
+        #endif
+
+        // Handle compressed weight
+        // ACCUMULATOR_TYPE max_weight = 0;
+        // USE_SLM == 1 && COMPRESSED_WEIGHTS_INT4 == 1
+        #if USE_SLM && COMPRESSED_WEIGHTS_INT4
+            #if TILE_OFM != 2
+            #error "FC bf_tiled kernel: can't use SLM optimization with TILE_OFM != 2"
+            #endif
+
+            // Skip first barrier synchronization if there is only single outer loop iteration.
+            #if MAIN_LOOP_ELEMENTS_COUNT / (TILE_IFM * SIMD) > 1
+                barrier(CLK_LOCAL_MEM_FENCE);
+            #endif
+
+            // Check scale
+            MAKE_VECTOR_TYPE(INPUT0_TYPE, TILE_B) scale_value_b = 1;
+            MAKE_VECTOR_TYPE(INPUT0_TYPE, TILE_B) r_scale_value_b = 1;
+            MAKE_VECTOR_TYPE(INPUT0_TYPE, TILE_B) max_input;
+            unroll_for (uint bi = 0; bi < TILE_B; ++bi) {
+                max_input[bi] = sub_group_reduce_max(fmax(fabs(in_0[bi][0]), fabs(in_0[bi][1])));
+            }
+            scale_value_b = max_input / 128;
+            r_scale_value_b = 1 / scale_value_b;
```

Comment:
Does it really make sense to have dynamic quantization within FC kernel? I think usually activation tensors are relatively small in LLMs, so probably running separate kernel which will quantize them won't have a big overhead. Have you tried something like that?

Reply:
Thanks for the comment. We will review that approach as well..